hdfs/client.py

```diff
@@ -661,7 +661,6 @@ def _download(_path_tuple):
             _writer.write(chunk)
 
     # First, we figure out where we will download the files to.
-    hdfs_path = self.resolve(hdfs_path)
```

Comment:
Rather than deleting this and introducing a new `_walk` function which will only be used for this use-case, could we cache the result of this call into another variable (e.g. `resolved_path`)?

Then we can just pass the original `hdfs_path` to `walk` below, without modifying the original function.